src/Simulation/Common/IQubitManager.cs

```diff
@@ -3,41 +3,37 @@
 
 namespace Microsoft.Quantum.Simulation.Common
 {
-    using Microsoft.Quantum.Simulation.Core;
     using System.Collections.Generic;
+    using Microsoft.Quantum.Simulation.Core;
 
     public interface IQubitManager
     {
+        bool DisableBorrowing { get; }
+
+        bool IsValid(Qubit qubit);
+        bool IsFree(Qubit qubit);
+        bool IsDisabled(Qubit qubit);
+
+        long FreeQubitsCount { get; }
+        long AllocatedQubitsCount { get; }
+        IEnumerable<long> AllocatedIds();
```

Comment:
if we switch the rest to be properties, why is this a method?

Reply:
The two counts require no computation; they are already fields on the implementation and Count properties are common for other data types. This one on the other hand likely requires computation and seems closer to the other methods.